File: 13. Rekursi Lanjut/B. Menggambar Pegunungan/b_menggambar_pegunungan.pas
program b_menggambar_pegunungan;

var
N : integer;

procedure draw(N : integer);
var
i : integer;
begin
	if (N = 1) then begin
		writeln('*');
	end
		else begin
		draw(N-1);
			for i := 1 to N do begin
			    write('*');
			end;
		writeln;
		draw(N-1);
		end;
end;

begin
    readln(N);
    draw(N);
    readln;
end.

File: 13. Rekursi Lanjut/B. Menggambar Pegunungan/b_menggambar_pegunungan.c
#include <stdio.h>

void draw(int N)
{
    if (N == 1)
    {
        printf("*\n");
    }
    else
    {
        draw(N - 1);
        int i;
        for (i = 0; i < N; i++)
        {
            printf("*");
        }
        printf("\n");
        draw(N - 1);
    }
}

int main()
{
    int N;
    scanf("%d", &N);
    draw(N);
    return 0;
}
